AssertionError: "since" timestamp of mv (1683879932799) is less than "since" timestamp of its source table (1683879933000)

### What version of Materialize are you using?

d5bd7d4691f7fb7340c69af84b5334c5615ffccf

### What is the issue?

Running `bin/mzcompose --release --find cluster run test-mv-source-sink` a few times locally I still sporadically get a failure with the materialized view having an older timestamp than the source table:
```
mzcompose: test case workflow-test-mv-source-sink failed:
Traceback (most recent call last):
  File "/home/deen/git/materialize2/misc/python/materialize/mzcompose/__init__.py", line 416, in test_case
    yield
  File "/home/deen/git/materialize2/misc/python/materialize/cli/mzcompose.py", line 589, in handle_composition
    composition.workflow(args.workflow, *args.unknown_subargs[1:])
  File "/home/deen/git/materialize2/misc/python/materialize/mzcompose/__init__.py", line 332, in workflow
    func(self)
  File "/home/deen/git/materialize2/test/cluster/mzcompose.py", line 1475, in workflow_test_mv_source_sink
    assert (
AssertionError: "since" timestamp of mv (1683880144289) is less than "since" timestamp of its source table (1683880145000)
```
I modified my change to make it fail reliably in CI too: https://github.com/MaterializeInc/materialize/pull/19238/commits/9e9f895426b3ca9912664a62f41d0e9318b5d0a0 Edit: My attempts to make it reproduce in CI failed, but locally it's easy, so that's strange.

Follow-up to https://github.com/MaterializeInc/materialize/issues/19151

--- a/test/cluster/mzcompose.py
+++ b/test/cluster/mzcompose.py
@@ -1456,8 +1456,6 @@ def workflow_test_mv_source_sink(c: Composition) -> None:
             WORKERS 2
         ));
         SET cluster = cluster1;
-        CREATE TABLE t (a int);
-        CREATE MATERIALIZED VIEW mv AS SELECT * FROM t;
         """
     )
 
@@ -1466,11 +1464,15 @@ def workflow_test_mv_source_sink(c: Composition) -> None:
         (since,) = j["determination"]["since"]["elements"]
         return int(since)
 
+    cursor = c.sql_cursor()
+    cursor.execute("CREATE TABLE t (a int)")
     # Verify that there are no empty frontiers.
-    output = c.sql_query("EXPLAIN TIMESTAMP AS JSON FOR SELECT * FROM t")
-    t_since = extract_since_ts(output[0][0])
-    output = c.sql_query("EXPLAIN TIMESTAMP AS JSON FOR SELECT * FROM mv")
-    mv_since = extract_since_ts(output[0][0])
+    cursor.execute("EXPLAIN TIMESTAMP AS JSON FOR SELECT * FROM t")
+    t_since = extract_since_ts(cursor.fetchall()[0][0])
+
+    cursor.execute("CREATE MATERIALIZED VIEW mv AS SELECT * FROM t")
+    cursor.execute("EXPLAIN TIMESTAMP AS JSON FOR SELECT * FROM mv")
+    mv_since = extract_since_ts(cursor.fetchall()[0][0])
 
     assert (
         mv_since >= t_since

